File: Task3/Haskell/inoutdiamond.hs
import System.IO

main :: IO ()
main = do
    input <- readFile "input.txt"
    let n = read (input :: String) :: Int
        topHalf = [generateRow n i | i <- [1..n]]
        bottomHalf = [generateRow n i | i <- [n-1, n-2..1]]
        diamond = unlines (topHalf ++ bottomHalf)
    writeFile "output.txt" diamond

generateRow :: Int -> Int -> String
generateRow n i = replicate (n - i) ' ' ++ replicate (2 * i - 1) '*'

File: Task3/Rust/inoutdiamond.rs
use std::fs;
use std::io::Result;

fn main() -> Result<()> {
    let input = fs::read_to_string("input.txt")?;
    let n: usize = input.trim().parse().expect("Invalid number");

    let mut diamond = String::new();

    for i in 1..=n {
        diamond.push_str(&format!("{:width$}\n", "*".repeat(2 * i - 1), width = n - i + 1));
    }

    for i in (1..n).rev() {
        diamond.push_str(&format!("{:width$}\n", "*".repeat(2 * i - 1), width = n - i + 1));
    }

    fs::write("output.txt", diamond)?;
    Ok(())
}

